bundles/org.eclipse.equinox.p2.engine/src/org/eclipse/equinox/internal/p2/engine/phases/AuthorityChecker.java

```diff
@@ -270,16 +278,29 @@ public static void gatherCertificates(Map<URI, List<Certificate>> authorities, I
 							var peerCertificates = sslSession.getPeerCertificates();
 							entry.getValue().addAll(Arrays.asList(peerCertificates));
 						} catch (SSLPeerUnverifiedException e) {
-							//$FALL-THROUGH$
+							LogHelper.log(new Status(IStatus.WARNING, EngineActivator.ID,
+									Messages.AuthorityChecker_GatherCertificatesFailure, e));
 						}
 					});
-				} catch (RuntimeException | InterruptedException | ExecutionException e) {
-					//$FALL-THROUGH$
+				} catch (InterruptedException e) {
+					Thread.currentThread().interrupt();
+				} catch (RuntimeException | ExecutionException e) {
+					LogHelper.log(new Status(IStatus.WARNING, EngineActivator.ID,
+							Messages.AuthorityChecker_GatherCertificatesFailure, e));
 				}
 			});
 		}
 	}
 
+	private static CompletableFuture<HttpResponse<String>> sendHttpRequestOrRetry(HttpClient client,
+			HttpRequest request, int retriesLeft) {
+		var future = client.sendAsync(request, BodyHandlers.ofString());
+		if (retriesLeft > 0) {
```

Comment:
Of course if we change the value from 2 to 3 above we should make sure to change this to respect that number.  It took me a minute to realize the like 297 creates on requested and that the recursive calls create two more.
```suggestion
		if (retriesLeft > 1) {
```